### ISSUE TITLE
File create issue when all drives have the same folder

### ISSUE THREAD

**Issue by ahowell82** [trapexit/mergerfs#358]
Problem: I cannot write a file both locally and via SMB to the union location
Further investigation:
I can write to a union location when there is only one drive that has the folder I am writing to.

Good write example: Disk 1 has the data folder where as disk 2 does not.  
Error write example: Disk 1 & 2 have the data folder

Configurations:
`/mnt/disk*  /mnt/storage     fuse.mergerfs defaults,allow_other,moveonenospc=true,fsname=snapraid,minfreespace=2G  0       0`

`user.mergerfs.srcmounts: /mnt/disk1:/mnt/disk2:/mnt/disk3:/mnt/disk4`
`user.mergerfs.minfreespace: 2147483648`
`user.mergerfs.moveonenospc: true`
`user.mergerfs.policies: all,epall,epff,eplfs,eplus,epmfs,eprand,erofs,ff,lfs,lus,mfs,newest,rand`
`user.mergerfs.version: 2.18.0-2-g10a9918`
`user.mergerfs.pid: 420`
`user.mergerfs.category.action: all`
`user.mergerfs.category.create: epmfs`
`user.mergerfs.category.search: ff`
`user.mergerfs.func.access: ff`
`user.mergerfs.func.chmod: all`
`user.mergerfs.func.chown: all`
`user.mergerfs.func.create: epmfs`
`user.mergerfs.func.getattr: ff`
`user.mergerfs.func.getxattr: ff`
`user.mergerfs.func.link: all`
`user.mergerfs.func.listxattr: ff`
`user.mergerfs.func.mkdir: epmfs`
`user.mergerfs.func.mknod: epmfs`
`user.mergerfs.func.open: ff`
`user.mergerfs.func.readlink: ff`
`user.mergerfs.func.removexattr: all`
`user.mergerfs.func.rename: all`
`user.mergerfs.func.rmdir: all`
`user.mergerfs.func.setxattr: all`
`user.mergerfs.func.symlink: epmfs`
`user.mergerfs.func.truncate: all`
`user.mergerfs.func.unlink: all`
`user.mergerfs.func.utimens: all`

**Comment by rubylaser** [comment]
@trapexit is going to need to see the permissions of each of the underlying disks and the the folder structures as well as the smb.conf to have a chance to help you troubleshoot this issue.

**Comment by trapexit** [comment]
Yes. Please reproduce without SMB and provide a reproducable example.

**Comment by ahowell82** [comment]
Contents of `/etc/samba/smb.conf`:
`[global]`
`   workgroup = WORKGROUP`
`        server string = %h server (Samba, Ubuntu)`
`   dns proxy = no`
`   log file = /var/log/samba/log.%m`
`   max log size = 1000`
`   syslog = 0`
`   panic action = /usr/share/samba/panic-action %d`
`   server role = standalone server`
`   passdb backend = tdbsam`
`   obey pam restrictions = yes`
`   unix password sync = yes`
`   passwd program = /usr/bin/passwd %u`
`   passwd chat = *Enter\snew\s*\spassword:* %n\n *Retype\snew\s*\spassword:* %n\n *password\supdated\ssuccessfully* .`
`   pam password change = yes`
`   map to guest = bad user`
`   usershare allow guests = yes`
`[printers]`
`   comment = All Printers`
`   browseable = no`
`   path = /var/spool/samba`
`   printable = yes`
`   guest ok = no`
`   read only = yes`
`   create mask = 0700`
`[print$]`
`   comment = Printer Drivers`
`   path = /var/lib/samba/printers`
`   browseable = yes`
`   read only = yes`
`   guest ok = no`
`[Storage]`
`comment =  Long term storage`
`path = /mnt/storage`
`valid users = adam`
`read only = no`
`create mask = 0777`

**Comment by ahowell82** [comment]
File permissions:
`adam@ubuntu:/mnt$ ls -la disk*`
`disk1:`
`total 9`
`drwxrwxrwx 1 root root 4096 Jan 15 23:38 .`
`drwxrwxrwx 7 root root 4096 Jan 15 22:01 ..`
`-rwxrwxrwx 1 root root  356 Jan 15 23:34 content`
`-rwxrwxrwx 1 root root    0 Jan 15 23:38 content.lock`
`drwxrwxrwx 1 root root    0 Dec 29 20:32 RAIDMain`
`drwxrwxrwx 1 root root    0 Jan 15 22:35 Unprotected`
``
`disk2:`
`total 9`
`drwxrwxrwx 1 root root 4096 Jan 15 23:38 .`
`drwxrwxrwx 7 root root 4096 Jan 15 22:01 ..`
`-rwxrwxrwx 1 root root  356 Jan 15 23:34 content`
`drwxrwxrwx 1 root root    0 Jan 15 22:34 RAIDMain`
`drwxrwxrwx 1 root root    0 Jan 15 23:59 Unprotected`
``
`disk3:`
`total 8`
`drwxrwxrwx 1 root root 4096 Jan 16 03:21 .`
`drwxrwxrwx 7 root root 4096 Jan 15 22:01 ..`
``
`disk4:`
`total 9`
`drwxrwxrwx 1 root root 4096 Jan 15 23:38 .`
`drwxrwxrwx 7 root root 4096 Jan 15 22:01 ..`
`-rwxrwxrwx 1 root root  356 Jan 15 23:34 content`
`drwxrwxrwx 1 root root    0 Jan 15 22:34 RAIDMain`
`drwxrwxrwx 1 root root    0 Jan 16 00:01 Unprotected`

**Comment by ahowell82** [comment]
Below is the error I receive when I attempt to touch a file.  However, I receive no error when I go to the mounted locations.

`adam@ubunutu:/mnt/storage$ touch this`
`touch: setting times of 'this': No such file or directory`

**Comment by trapexit** [comment]
What does Samba have to do with this?

Can you please create a minimal example of this rather than providing me with your system setup? 

For example I just did:

```
$ sudo mkdir /tmp/{a,b,c}
$ sudo chmod ugo=rwx /tmp/{a,b,c}
$ sudo mergerfs -odefaults,allow_other,moveonenospace=true,minfreespace=2G /tmp/a:/tmp/b /tmp/c
$ touch /tmp/c/foo
```

And everything worked.

**Comment by ahowell82** [comment]
I started to look at the drive formatting & noticed that 3 of my drives were NFTS & one was ext4.  I decided to see if there was one drive that was goofy & making problems for the rest.  after running through the permutations.  I decided to reformat the drives as ext4; i received a udisks-error-quark on one of the drives. After some research, I found link to an askubuntu.com regarding the matter.  I installed gparted and did the formatting there.  I have mounted the drives and now things a working as I believe they should.

http://askubuntu.com/questions/377253/unable-to-format-usb-drive-with-disks-udisks-error-quark-0

Thank you all for your help,  I apologize for creating a ticket that should've been handled in some user help website
